django_projects/locallibrary/catalog/views.py

```diff
@@ -9,12 +9,15 @@ def index(request):
     num_instances = BookInstance.objects.all().count()
     num_instances_available = BookInstance.objects.filter(status__exact='a').count()
     num_authors = Author.objects.count()
+    num_visits = request.session.get('num_visits', 0)
```

Comment:
method dưới hoạt động như nào e nhỉ ?
```suggestion
    get('num_visits', 0)
```

Reply:
Đầu tiên đặt giá trị khóa session bằng 0, mỗi lần nhận request từ client truy cập trang đó, thì tăng giá trị và lưu trữ trong session. Sau đó biến num_visits được chuyển thành context hiển thị trên view html

Reply:
Đầu tiên đặt giá trị khóa session bằng 0, mỗi lần nhận request từ client truy cập trang đó, thì tăng giá trị và lưu trữ trong session. Sau đó biến num_visits được chuyển thành context hiển thị trên view html

Reply:
Chị hỏi method get thôi thì ở đây là get value trong session với key 'num_visits'
- nếu tìm thấy thì trả về value tương ứng của key đó
- nếu tìm không có thì trả về giá trị mặc định là 0